Code/client/ModCompat/BehaviorVar.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+#ifdef MODDED_BEHAVIOR_COMPATIBILITY
+
+#include <Structs/AnimationGraphDescriptorManager.h>
+
+struct BehaviorVar
+{
+    struct Replacer
+    {
+        uint64_t origHash;
+        uint64_t newHash;
+        std::string signatureVar;
+        std::string creatureName;
+        std::vector<std::string> syncBooleanVar;
+        std::vector<std::string> syncFloatVar;
+        std::vector<std::string> syncIntegerVar;
+        bool operator==(const uint64_t& acRHS) const
+        {
+            return origHash == acRHS;
+        }
+    };
+
+    static BehaviorVar* Get();
+    const AnimationGraphDescriptor* Patch(BSAnimationGraphManager* apManager, Actor* apActor);
+    boolean failListed(const uint64_t acHash);
+    void failList(const uint64_t acHash);
+
+    void Init();
+    void Debug();
+
+  private:
+    static BehaviorVar* single;
+    uint64_t invocations = 0;
+
+    void seedAnimationVariables(
+        const uint64_t acHash, 
+        const AnimationGraphDescriptor* acpDescriptor, 
+        const std::map<const std::string, const uint32_t>& acReversemap,
+        std::set<uint32_t>& aBoolVars, 
+        std::set<uint32_t>& aFloatVars,
+        std::set<uint32_t>& aIntVars);
+
+    const std::vector<std::string> tokenizeBehaviorSig(const std::string acSignature) const;
+    const AnimationGraphDescriptor* constructModdedDescriptor(
+        const uint64_t acNewHash, const Replacer& acReplacer,
+        std::map<const std::string, const uint32_t>& acReverseMap);
+
+    std::vector<std::filesystem::path> loadDirs(const std::filesystem::path& acPATH);
+    Replacer* loadReplacerFromDir(const std::filesystem::path acDir);
+    std::vector<uint64_t> signatureMatches(const uint64_t acHash, const std::string acSignature) const;
+
+    std::vector<Replacer> behaviorPool; // Pool for loaded behaviours
```

Comment:
Tiltedcore vector

Reply:
Commit 2d989d98047dfa179045b7ba45333d5f17575a1b

Fixed it. It's a pretty narrow window, though. The path is from a directory iterator just run; so either the file was just deleted in the last millisecond or the directory is read-protected; a lot more would break in the latter case.